pkg/workqueue/inmem/inmem.go

```diff
@@ -64,6 +67,36 @@ func (w *wq) Queue(_ context.Context, key string, opts workqueue.Options) error
 	return nil
 }
 
+// Get implements workqueue.Interface.
+func (w *wq) Get(_ context.Context, key string) (*workqueue.KeyState, error) {
+	w.rw.RLock()
+	defer w.rw.RUnlock()
+
+	if _, ok := w.wip[key]; ok {
+		return &workqueue.KeyState{
+			Key:    key,
+			Status: workqueue.KeyState_IN_PROGRESS,
+		}, nil
+	}
+
+	if qi, ok := w.queue[key]; ok {
+		attempts := qi.attempts
+		if attempts > math.MaxInt32 {
+			attempts = math.MaxInt32
+		}
+		return &workqueue.KeyState{
+			Key:           key,
+			Status:        workqueue.KeyState_QUEUED,
+			Priority:      qi.Priority,
+			Attempts:      int32(attempts), //nolint:gosec  // we're not worried about this overflowing
+			QueuedTime:    qi.queued.Unix(),
+			NotBeforeTime: qi.NotBefore.Unix(),
+		}, nil
+	}
```

Comment:
Does in-memory not have DLQ? 🤔

Reply:
It has a deadletter func, but it just removes it. It doesn't persist like the GCS implementation does.